app/controllers/comments_controller.rb

```diff
@@ -20,4 +21,11 @@ def comment_params
     params.require(:comment).permit(:text).merge(user_id: current_user.id)
   end
   
+  def correct_user
+    @comment = Comment.find(params[:id])
+    if @comment.user != current_user
+      redirect_to root_url
+    end
+  end
```

Comment:
Looks like you made an edit here so my previous comment went away. This method should be private. Also, no need to use instance variables anywhere in this method.